browser_patches/webkit/patches/bootstrap.diff

```diff
@@ -4906,6 +5045,119 @@ index fc6c5f3cd046269566822c08a482cc3bd6a883c0..083c03903a3111ee2cb6f9882dd2efe1
      platform/graphics/cairo/PlatformContextCairo.h
      platform/graphics/cairo/RefPtrCairo.h
  )
+diff --git a/Source/WebCore/platform/DragData.h b/Source/WebCore/platform/DragData.h
+index 4e53e633374861fd3171285b5c5e91f681023e0b..e665b5d218808b8cf69f45e48bc1bb7ba659a9d3 100644
+--- a/Source/WebCore/platform/DragData.h
++++ b/Source/WebCore/platform/DragData.h
+@@ -47,7 +47,7 @@ typedef void* DragDataRef;
+ 
+ #elif PLATFORM(WIN)
+ typedef struct IDataObject* DragDataRef;
+-#elif PLATFORM(GTK)
++#elif PLATFORM(GTK) || PLATFORM(WPE)
+ namespace WebCore {
+ class SelectionData;
+ }
+@@ -112,7 +112,7 @@ public:
+     bool containsPromise() const;
+ #endif
+ 
+-#if PLATFORM(GTK)
++#if PLATFORM(GTK) || PLATFORM(WPE)
+ 
+     DragData& operator =(const DragData& data)
+     {
+diff --git a/Source/WebCore/platform/DragImage.cpp b/Source/WebCore/platform/DragImage.cpp
+index a824791c20f4928bb45473ea05d33d0afc3825e2..c11b7d454b42b549be36f7122290fc9ed905734a 100644
+--- a/Source/WebCore/platform/DragImage.cpp
++++ b/Source/WebCore/platform/DragImage.cpp
+@@ -272,7 +272,7 @@ DragImage::~DragImage()
+         deleteDragImage(m_dragImageRef);
+ }
+ 
+-#if !PLATFORM(COCOA) && !PLATFORM(GTK) && !PLATFORM(WIN)
++#if !PLATFORM(COCOA) && !PLATFORM(GTK) && !PLATFORM(WIN) && !PLATFORM(WPE)
+ 
+ IntSize dragImageSize(DragImageRef)
+ {
+diff --git a/Source/WebCore/platform/Pasteboard.h b/Source/WebCore/platform/Pasteboard.h
+index aadc46698df79564362a03f5dfe89b0eb7811ce9..4a143b1b884d920a0140cb5821738715205302c7 100644
+--- a/Source/WebCore/platform/Pasteboard.h
++++ b/Source/WebCore/platform/Pasteboard.h
+@@ -43,7 +43,7 @@ OBJC_CLASS NSString;
+ OBJC_CLASS NSArray;
+ #endif
+ 
+-#if PLATFORM(GTK)
++#if PLATFORM(GTK) || PLATFORM(WPE)
+ #include "SelectionData.h"
+ #endif
+ 
+@@ -90,16 +90,12 @@ struct PasteboardWebContent {
+     Vector<String> clientTypes;
+     Vector<RefPtr<SharedBuffer>> clientData;
+ #endif
+-#if PLATFORM(GTK)
++#if PLATFORM(GTK) || PLATFORM(WPE)
+     String contentOrigin;
+     bool canSmartCopyOrDelete;
+     String text;
+     String markup;
+ #endif
+-#if USE(LIBWPE)
+-    String text;
+-    String markup;
+-#endif
+ };
+ 
+ struct PasteboardURL {
+@@ -108,7 +104,7 @@ struct PasteboardURL {
+ #if PLATFORM(MAC)
+     String userVisibleForm;
+ #endif
+-#if PLATFORM(GTK)
++#if PLATFORM(GTK) || PLATFORM(WPE)
+     String markup;
+ #endif
+ };
+@@ -181,11 +177,11 @@ public:
+ 
+ #if PLATFORM(GTK)
+     explicit Pasteboard(const String& name);
++#endif
++#if (PLATFORM(WPE) || PLATFORM(GTK)) && ENABLE(DRAG_SUPPORT)
```

Comment:
It seems that we can simply add || PLATFORM(WPE) to the condition above (just for the sake of reducing the diff).

Reply:
It was a little tricky to merge the two, but I just left this alone and added extra declarations that should be easier to diff against.